code_generator/GdkCodeGenerator/Templates/UnityReaderWriterGenerator.tt

```diff
@@ -23,44 +23,48 @@ namespace <#= qualifiedNamespace #>
         [ComponentId(<#= unityComponentDefinition.Id #>)]
         internal class ReaderWriterCreator : IReaderWriterCreator
         {
-            public IReaderInternal CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
+            public IReaderWriterInternal CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
             {
                 return new ReaderWriterImpl(entity, entityManager, logDispatcher);
             }
         }
 
         [ReaderInterface]
         [ComponentId(<#= unityComponentDefinition.Id #>)]
-        public interface Reader
+        public interface Reader : IReader<<#= componentDetails.TypeName #><# if(fieldDetailsList.Count > 0){ #>, <#= componentDetails.TypeName #>.Update<# } #>>
```

Comment:
Fine for this PR - but I wonder if there's a more elegant way of writing this

Reply:
I have an idea :)